# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ModuleLoaderBase.h
# Commit: 22649272ffae
# Full Hash: 22649272ffae2ec3077c452cf5acce0326758623
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-18 21:52:29
# Regressor Bug: 1742437
# File Overlap Count: 1
# Description:
#   Bug 1742437 - move DOM module loader into own file; r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D137531
# ==============================================================================

diff -r e3fe772947bf -r 22649272ffae dom/script/ModuleLoaderBase.h
--- a/dom/script/ModuleLoaderBase.h	Fri Feb 18 12:45:51 2022 +0000
+++ b/dom/script/ModuleLoaderBase.h	Fri Feb 18 12:45:51 2022 +0000
@@ -86,7 +86,9 @@
       mozilla::MaybeOneOf<JS::SourceText<char16_t>, JS::SourceText<Utf8Unit>>;
 
   // Methods that must be overwritten by an extending class
-  virtual void EnsureModuleHooksInitialized() = 0;
+  virtual void EnsureModuleHooksInitialized() {
+    MOZ_ASSERT(false, "You must override EnsureModuleHooksInitialized");
+  }
   virtual nsresult StartModuleLoad(ScriptLoadRequest* aRequest) = 0;
   virtual void ProcessLoadedModuleTree(ModuleLoadRequest* aRequest) = 0;
   virtual nsresult CompileOrFinishModuleScript(